cinnabot.go

```diff
@@ -132,12 +143,72 @@ func (cb *Cinnabot) Router(msg tgbotapi.Message) {
 		cb.log.Printf("[%s][id: %d] command: %s, args: %s", time.Now().Format(time.RFC3339), cmsg.MessageID, cmsg.Cmd, cmsg.GetArgString())
 	}
 	execFn := cb.fmap[cmsg.Cmd]
-
+	log.Print(msg.Chat.ID)
 	if execFn != nil {
 		cb.GoSafely(func() { execFn(cmsg) })
-	} else {
-		cb.SendTextMessage(msg.From.ID, "No such command!")
+
+		cb.cache.Set(strconv.Itoa(msg.From.ID), cmsg.Cmd, cache.DefaultExpiration)
+
+	} else if cmdRaw, check := cb.cache.Get(strconv.Itoa(msg.From.ID)); check {
+		//Have to typecast
+		cmd := cmdRaw.(string)
+		//If cmd in cache and arg matches command
+
+		cmsg.Args = append([]string{cmsg.Cmd}, cmsg.Args...)
+
+		if cb.CheckArgCmdPair(cmd, cmsg.Args) {
+			//Get function from previous command
+			execFn = cb.fmap[cmd]
+			//Ensure tokens is in order [unecessary]
+			cmsg.Cmd = cmd
+
+			cb.GoSafely(func() { execFn(cmsg) })
+			return
+		}
+		log.Print("Out")
+		log.Print(cmd)
+		replyMessage := tgbotapi.NewMessage(int64(msg.From.ID), "No such command!")
+		replyMessage.ReplyMarkup = tgbotapi.NewRemoveKeyboard(true)
+		cb.SendMessage(replyMessage)
+
+	}
+}
+
+//Checks if arg can be used with command
+//Used to supplement cache as cache only records functions as states
+func (cb *Cinnabot) CheckArgCmdPair(cmd string, args []string) bool {
```

Comment:
cache automatically stores the last used command for the user in cinnabot. It can also be hijacked in any function to declare what the last used command is. 
An example is in feedback after the user selects an option. The feedback command will register the last command as "/cinnabotFeedback" if the user selected cinnabot. Subsequent messages are then registered as arguments to the "/cinnabotFeedback" command. 

I then have to be able to decide when these arguments are valid arguments to the command. /nusbus for example, only accepts cinnamon or <blank> as valid arguments. (Because location gives a blank message). cb.Router then reads the argument and then executes the function if the pair is valid, else it treats it as though you were sending gibberish.